src/app/data-planes/components/DataplanePolicies.vue

```diff
@@ -161,44 +188,128 @@ function getPolicyRoutes(policies: Record<string, PolicyType> | undefined): Mesh
+    let sourceTags: LabelValue[] | null
+    let destinationTags: LabelValue[] | null
+
+    if (type === 'clientSubset') {
+      // Sets the wildcard service tag for client subsets without subset values.
+      const sourceSubset = subsetEntries.length > 0 ? subsetEntries : [['kuma.io/service', '*']]
+      // For client subsets, the source is represented by `subset` (i.e. tags); for destination subsets and single items, the source is the DPP.
+      sourceTags = sourceSubset.length > 0 ? sourceSubset.map(([label, value]) => ({ label, value })) : null
+    }
 
-      processedMatchedPolicies.push({
-        name: policyType,
-        pluralName: policyDefinition.pluralDisplayName,
-        policies,
+    if (typeof service === 'string' && service !== '') {
+      const destinationSubset = [['kuma.io/service', service]]
+      destinationTags = destinationSubset.length > 0 ? destinationSubset.map(([label, value]) => ({ label, value })) : null
+    } else if (type === 'destinationSubset') {
+      // Sets the wildcard service tag for destination subsets without subset values.
+      const destinationSubset: [string, string][] = subsetEntries.length > 0 ? subsetEntries : [['kuma.io/service', '*']]
+      // For client subsets and destination subsets, the destination is `service`; for single items, the destination is the DPP.
+      destinationTags = destinationSubset.length > 0 ? destinationSubset.map(([label, value]) => ({ label, value })) : null
+    }
+
+    const name = type === 'clientSubset' || type === 'destinationSubset' || service ? rule.name : null
+    const config = conf && Object.keys(conf).length > 0 ? json2yaml(JSON.stringify(conf, null, 2)) : null
+
+    if (!policyEntriesByType.has(rule.policyType)) {
+      policyEntriesByType.set(rule.policyType, {
+        type: rule.policyType,
+        connections: [],
       })
     }
 
-    const { name, type, service } = sidecarDataplane
-    dataplanePolicies.push({
-      name,
-      type,
-      service,
-      matchedPolicies: processedMatchedPolicies,
-    })
+    const policyEntry = policyEntriesByType.get(rule.policyType) as PolicyTypeEntry
+    const policyDefinition = store.state.policiesByType[rule.policyType]
+    const origins: PolicyTypeEntryOrigin[] = []
+
+    for (const ruleOrigin of rule.origins) {
+      origins.push({
+        name: ruleOrigin.name,
+        route: {
+          name: policyDefinition.path,
+          query: { ns: ruleOrigin.name },
+          params: { mesh: ruleOrigin.mesh },
+        },
+      })
+    }
+
+    const connection: PolicyTypeEntryConnection = { sourceTags, destinationTags, name, config, origins }
+    policyEntry.connections.push(connection)
   }
 
-  return dataplanePolicies
+  const policyTypeEntries = Array.from(policyEntriesByType.values())
+
+  policyTypeEntries.sort((policyEntryA, policyEntryB) => policyEntryA.type.localeCompare(policyEntryB.type))
+
+  return policyTypeEntries
```

Comment:
I found all of this a bit 'intense' to look at, I can see that its mostly reshaping data for easier lookup (I assume). I had two things I was wondering, is it worth organising it a little more to make it unit testable? Or another idea was adding some more comments in here just giving a brief explanation of what's happening and why. Both of which would probably help when wondering what shape we are going from > to. Appreciate you are working with the code thats already there so maybe no is not the right time? Thought I'd ask in case.

I'm also getting some editor errors down on line 304 assuming because sourceTags and destinationTags can be `undefined` rather than `null`, are you getting those? Not a biggie

Reply:
> I process the data in such a way that it fits the use case of respective templates as close as possible to avoid doing bigger chunks of logic in templates.

Yeah for sure, makes sense 👍 

> This process is inadvertently complex

Yeah that's why I wondered if unit test might have helped me (no action here by the way if you are happy with how it is)

> It would help to know which parts are particularly hard to follow.

It wasn't anything in particular, just a general "looks like theres a lot of stuff going on here", might be partly down to my unfamiliarity with the data also.

Reply:
I agree there is a lot going on with this. I just can’t think of something clever to comment that doesn’t amount to "well basically that one child component needs it to look like that"